### PR TITLE
vendor: bump boltdb to v1.2.1

### PR DESCRIPTION
**Pull request by cyphar**
This includes a fix to soft database corruption that would cause Docker
to fail to start if the daemon died in the middle of a transaction
write.

Credit Photo (creative commons): Sam W

Signed-off-by: Aleksa Sarai asarai@suse.de

### CODE REVIEW

File: hack/vendor.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ clone git github.com/coreos/etcd v2.2.0
 fix_rewritten_imports github.com/coreos/etcd
 clone git github.com/ugorji/go 5abd4e96a45c386928ed2ca2a7ef63e2533e18ec
 clone git github.com/hashicorp/consul v0.5.2
-clone git github.com/boltdb/bolt v1.2.0
+clone git github.com/boltdb/bolt d97499360d1ecebc492ea66c7447ea948f417620
```
Comment (thaJeztah) [inline hack/vendor.sh]: Any chance they create a new tag for this?

Comment (cyphar) [inline hack/vendor.sh]: They didn't, I can ask.

Comment (thaJeztah) [inline hack/vendor.sh]: Thanks!

Comment (benbjohnson) [inline hack/vendor.sh]: I added a tagged release ([v1.2.1](https://github.com/boltdb/bolt/releases/tag/v1.2.1)).

**Comment by benbjohnson** [comment]
One additional note for posterity. The failure only occurs on a partial sector write to a meta page (which is extremely rare) and not simply any transaction failure. This is actually the only time I've ever heard of this occurring for someone using Bolt.

**Comment by cyphar** [comment]
@benbjohnson Sorry, I didn't mean to make it sound worse than it was. But it did cause bolt to fail Docker from starting (and it's very reproducible if you put the Docker daemon under heavy load such that the kernel would kill the process on a `malloc()`).

**Comment by benbjohnson** [comment]
@cyphar That's interesting that it would cause a partial sector write. I would think that the kernel would still write a full sector even if the process dies. I'm glad it's fixed now though! :)

File: hack/vendor.sh (hunk 1 above)
Comment (thaJeztah) [inline hack/vendor.sh]: Thanks @benbjohnson!

**Comment by runcom** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM

CI looks to have a bad day

**Comment by mavenugo** [comment]
LGTM.
Though not mandatory for this PR, it would be good to bump up the boltdb Godeps in `docker/libnetwork` as well to be consistent.

**Comment by LK4D4** [comment]
pretty fun failure on windows
ping @jhowardmsft

**Comment by jhowardmsft** [comment]
@LK4D4 You want @tiborvass for the win2lin - error is coming from the Linux side.

```
00:00:45.300 Cloning into '/tmp/tmp.4T97ICpxUb/src/github.com/docker/notary'...
00:10:46.955 fatal: unable to access 'https://github.com/docker/notary.git/': GnuTLS recv error (-54): Error in the pull function.
```

**Comment by LK4D4** [comment]
@jhowardmsft oh, I meant TP5
```
17:14:01 FAIL: docker_cli_run_test.go:2336: DockerSuite.TestRunUnknownCommand
17:14:01 
17:14:01 docker_cli_run_test.go:2356:
17:14:01     c.Fatalf("ExitCode(%v) cannot be 0", rc)
17:14:01 ... Error: ExitCode(0) cannot be 0
```

**Comment by jhowardmsft** [comment]
I've seen that on two PRs now. Not sure if something got merged into master, or is a flakey test. Will investigate when I get a chance.

**Comment by thaJeztah** [comment]
Created an issue and marked it as "flaky test" for now; https://github.com/docker/docker/issues/22795

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**